pkg/roachpb/api.go

```diff
@@ -966,3 +966,23 @@ func (b *BulkOpSummary) Add(other BulkOpSummary) {
 	b.IndexEntries += other.IndexEntries
 	b.SystemRecords += other.SystemRecords
 }
+
+// Equal method is just used to verify whether two ComputeChecksumRequest is equal.
+func (ccr *ComputeChecksumRequest) Equal(cr *ComputeChecksumRequest) bool {
```

Comment:
Because if we generate `Equal` method like this:

```
func (this *ComputeChecksumRequest) Equal(that interface{}) bool {
	if that == nil {
		if this == nil {
			return true
		}
		return false
	}

	that1, ok := that.(*ComputeChecksumRequest)
	if !ok {
		that2, ok := that.(ComputeChecksumRequest)
		if ok {
			that1 = &that2
		} else {
			return false
		}
	}
	if that1 == nil {
		if this == nil {
			return true
		}
		return false
	} else if this == nil {
		return false
	}
	if !this.Span.Equal(&that1.Span) {
		return false
	}
	if this.Version != that1.Version {
		return false
	}
	if !this.ChecksumID.Equal(that1.ChecksumID) {
		return false
	}
	if this.Snapshot != that1.Snapshot {
		return false
	}
	return true
}
```

we have already define a `Equal` method for `Span` struct, just like:

```
// Equal compares for equality.
func (s Span) Equal(o Span) bool {
	return s.Key.Equal(o.Key) && s.EndKey.Equal(o.EndKey)
}
```

so if will specify the `option (gogoproto.equal) = true;` for `message Span` in `data.proto`, when we `make build`, it will report an error:
```
pkg/roachpb/api.pb.go:1892: cannot use &that1.Span (type *Span) as type Span in argument to this.Span.Equal
pkg/roachpb/data.pb.go:552: method redeclared: Span.Equal
	method(Span) func(Span) bool
	method(*Span) func(interface {}) bool
```